pkg/controller/servicebindingrequest/reconciler.go

```diff
@@ -194,6 +203,33 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		return sb.Unbind()
 	}
 
+	var emptyApplication v1alpha1.ApplicationSelector
+	if sbr.Spec.ApplicationSelector == emptyApplication ||
+		(sbr.Spec.ApplicationSelector.ResourceRef == "" &&
+			sbr.Spec.ApplicationSelector.LabelSelector.MatchLabels == nil) {
+		_, updateErr := updateServiceBindingRequestStatus(
+			r.dynClient,
+			sbr,
+			[]conditionsv1.Condition{
+				conditionsv1.Condition{
+					Type:    InjectionReady,
+					Status:  corev1.ConditionFalse,
+					Reason:  EmptyApplicationSelectorReason,
+					Message: "The spec.applicationSelector field is empty.",
+				},
+				conditionsv1.Condition{
+					Type:   CollectionReady,
+					Status: corev1.ConditionTrue,
+				},
+			}...,
+		)
+		if updateErr != nil {
+			logger.Error(err, "Error updating service-binding-request status.")
+			return RequeueError(updateErr)
```

Comment:
Shall we requeue error if application is missing? Because it is collected so I believe we should skip and return Done() and above in empty backingserviceSelector we can RequeueError(updateErr) because if there is no backing service then there is no complete binding and no generic binding, i.e no collection and injection ready

Reply:
If there is an error while updating status that would never be reflected on the SBR.

Reply:
A Service Binding Request should always be re-queued if the declared selector results in an empty resource set from k8s.

Reply:
@isutton 

In this case, the `applicationSelector` is not declared by the user in YAML.
So, I return a `Done()` -> https://github.com/isutton/service-binding-operator/pull/13/files#diff-42835fd3b25b196b1bbc21495e0ac5b7R241

When the user updates the YAML with applicationSelector, the reconcilation happens again to fulfill the request.

If the declared application is not found, THEN Service Binding Request should be re-queued.

------------------------------------------------------------------------------------------------------

However, when there is **an error in updating the status** of Service Binding Request -> https://github.com/isutton/service-binding-operator/pull/13/files#diff-42835fd3b25b196b1bbc21495e0ac5b7R237

then re-queue should happen.

Reply:
Avni, Igor do you agree with this?

Reply:
Yes.

Summing up:

* empty `applicationSelector` should not re-queue the service binding resource
* valid `applicationSelector` should re-queue the service binding resource in case the k8s api server returns an empty list of resources, indicating the user expects resources to exist at some point

Reply:
Highlighting this again,

```
return RequeueError(updateErr)
```

This requeue is only if there is an error while updating the status of the SBR.